ios/RCTMapboxGL/RCTMapboxGL.m

```diff
@@ -469,6 +469,39 @@ - (void)handleLongPress:(UITapGestureRecognizer *)sender
     }
 }
 
+- (void)mapViewDidFinishLoadingMap:(MGLMapView *)mapView
+{
+    NSDictionary *event = @{ @"target": self.reactTag,
+                             @"src": @{
+                                     @"loaded": @YES
+                                     }
+                             };
+    
+    [_eventDispatcher sendInputEventWithName:@"onFinishLoadingMap" body:event];
+}
+- (void)mapViewWillStartLoadingMap:(MGLMapView *)mapView
+{
+    NSDictionary *event = @{ @"target": self.reactTag,
+                             @"src": @{
+                                     @"loading": @YES
+                                     }
+                             };
+    
+    [_eventDispatcher sendInputEventWithName:@"onStartLoadingMap" body:event];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFailToLocateUserWithError:(NSError *)error
+{
+    NSDictionary *event = @{ @"target": mapView.reactTag,
+                             @"src": @{
+                                     @"code": @(error.code),
```

Comment:
`code` is one of the enum values [listed here](https://developer.apple.com/library/mac/documentation/CoreLocation/Reference/CoreLocationConstantsRef/#//apple_ref/c/tdef/CLError). Are developers expected to compare the code to (say) `2` to discover a network issue, or does React Native automatically expose the `CLError` values to the JS side as well?